apex_launchtest/apex_launchtest/asserts/assert_sequential_output.py

```diff
@@ -50,7 +50,43 @@ def assertInStdout(self, msg):
             array_index += 1
             substring_index = 0
 
-        assert False, "{} not found in output".format(msg)
+        assert False, (
+            "'{}' not found in sequence after previous match.  "
+            "The output near the last matched line:\n{}"
+        ).format(
+            msg,
+            self.get_nearby_lines()
+        )
+
+    def get_nearby_lines(self):
+
+        # This works by concatinating a few of the process_io outputs that we received, then
+        # searching forward and backward for two return-lines in each direction, then returning
+        # just that portion to give context about where a failure happened
+
+        start_idx = max(0, self._array_index - 2)
+        end_idx = self._array_index + 4
```

Comment:
Wouldn't this print two lines before and four lines after if `start_idx > 2`?

Reply:
@jpsamper2009 
Since end_idx (on line 70 below) is exclusive, we need to use +4 so we get the current line plus the next two lines

Also, things get trimmed down a little before we return the final value.  First we turn a chunk of self._output into one long string, then we start looking forward and backward for '\n' characters.  One entry in self._output might contain multiple '\n' characters depending on how the IO buffers got flushed.